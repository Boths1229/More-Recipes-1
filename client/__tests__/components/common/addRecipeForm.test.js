import React from 'react';
import { shallow } from 'enzyme';
import AddRecipeForm from '../../../components/common/addRecipeForm';


let props;

const setup = () => {
  props = {
    errorMessage: '',
    handleSubmit: () => {},
    handleChange: () => {},
    onFocus: () => {},
    isUploading: false,
    progress: 0,
    handleProgress: () => {},
    handleUploadStart: () => {},
    handleUploadSuccess: () => {},
    notReady: true,
    mealType: '',
  };

  return shallow(<AddRecipeForm {...props} />);
};

describe('AddRecipeForm Component', () => {
  it('should render correctly', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('button').length).toBe(3);
    expect(wrapper.find('form').length).toBe(1);
  });

  it('should render error if error exists', () => {
    const wrapper = setup();
    wrapper.setProps({ errorMessage: 'select a meal type' });
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('div.alert').text()).toBe('select a meal type');
  });
});
