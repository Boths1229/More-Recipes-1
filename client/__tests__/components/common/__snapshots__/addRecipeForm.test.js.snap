// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipeForm Component should render correctly 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  data-backdrop="static"
  id="exampleModal"
  role="dialog"
  style={
    Object {
      "display": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="exampleModalLabel"
        >
          Add new recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <form
          id="recipeForm"
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Title:
            </label>
            <input
              className="form-control"
              id="recipe-name"
              name="recipeName"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Description:
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="image"
            >
              Image:
            </label>
            <[object Object]
              id="uploadImage"
              name="image"
              onProgress={[Function]}
              onUploadStart={[Function]}
              onUploadSuccess={[Function]}
              storageRef={
                Object {
                  "ref": [MockFunction] {
                    "calls": Array [
                      Array [
                        "images",
                      ],
                    ],
                  },
                  "storage": [MockFunction] {
                    "calls": Array [
                      Array [],
                    ],
                  },
                }
              }
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="type-text"
            >
              Type of meal:
            </label>
            <select
              className="form-control"
              id="type-text"
              name="mealType"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
              value=""
            >
              <option
                defaultValue="breakfast"
              >
                Breakfast
              </option>
              <option
                value="brunch"
              >
                Brunch
              </option>
              <option
                value="elevenses"
              >
                Elevenses
              </option>
              <option
                value="lunch"
              >
                Lunch
              </option>
              <option
                value="tea"
              >
                Tea
              </option>
              <option
                value="supper"
              >
                Supper
              </option>
              <option
                value="dinner"
              >
                Dinner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredients-text"
            >
              Ingredients:
            </label>
            <textarea
              className="form-control"
              id="ingredients"
              name="ingredients"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              placeholder="separate each ingredient with a comma"
              required={true}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="method-text"
            >
              Method of cooking:
            </label>
            <textarea
              className="form-control"
              id="method-text"
              name="method"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              placeholder="separate each method with a newline"
              required={true}
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn vote-button"
              type="submit"
            >
              Add recipe
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;

exports[`AddRecipeForm Component should render error if error exists 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  data-backdrop="static"
  id="exampleModal"
  role="dialog"
  style={
    Object {
      "display": "none",
    }
  }
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header black"
      >
        <h5
          className="modal-title black"
          id="exampleModalLabel"
        >
          Add new recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            ×
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        <div
          className="alert alert-danger"
        >
          select a meal type
        </div>
        <form
          id="recipeForm"
          onSubmit={[MockFunction]}
        >
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="recipient-name"
            >
              Title:
            </label>
            <input
              className="form-control"
              id="recipe-name"
              name="recipeName"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="message-text"
            >
              Description:
            </label>
            <textarea
              className="form-control"
              id="message-text"
              name="description"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="image"
            >
              Image:
            </label>
            <[object Object]
              id="uploadImage"
              name="image"
              onProgress={[Function]}
              onUploadStart={[Function]}
              onUploadSuccess={[Function]}
              storageRef={
                Object {
                  "ref": [MockFunction] {
                    "calls": Array [
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                      Array [
                        "images",
                      ],
                    ],
                  },
                  "storage": [MockFunction] {
                    "calls": Array [
                      Array [],
                      Array [],
                      Array [],
                    ],
                  },
                }
              }
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="type-text"
            >
              Type of meal:
            </label>
            <select
              className="form-control"
              id="type-text"
              name="mealType"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              required={true}
              value=""
            >
              <option
                defaultValue="breakfast"
              >
                Breakfast
              </option>
              <option
                value="brunch"
              >
                Brunch
              </option>
              <option
                value="elevenses"
              >
                Elevenses
              </option>
              <option
                value="lunch"
              >
                Lunch
              </option>
              <option
                value="tea"
              >
                Tea
              </option>
              <option
                value="supper"
              >
                Supper
              </option>
              <option
                value="dinner"
              >
                Dinner
              </option>
            </select>
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="ingredients-text"
            >
              Ingredients:
            </label>
            <textarea
              className="form-control"
              id="ingredients"
              name="ingredients"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              placeholder="separate each ingredient with a comma"
              required={true}
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="col-form-label black"
              htmlFor="method-text"
            >
              Method of cooking:
            </label>
            <textarea
              className="form-control"
              id="method-text"
              name="method"
              onChange={[MockFunction]}
              onFocus={[MockFunction]}
              placeholder="separate each method with a newline"
              required={true}
            />
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-secondary"
              data-dismiss="modal"
              type="button"
            >
              Close
            </button>
            <button
              className="btn vote-button"
              type="submit"
            >
              Add recipe
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
